### ISSUE TITLE
Revisit ReconnectNodeRemover synchronization

### ISSUE THREAD

**Issue by artemananiev** [hashgraph/hedera-services#9831]
This issue was found during investigation for https://github.com/hashgraph/hedera-services/issues/9764.

Current node remover threading model is as follows:

* LearnerThread receives all the nodes from the teacher
* On the same thread, nodes are deserialized and put to a) node remover b) reconnect hasher
* The remover doesn't handle the nodes right away, but puts them to a queue
* The queue is processed on a different thread
* Once LearnerThread is done with its main loop, the node remover is closed. Some elements may still be unprocessed in the queue
* In the fix for #9764 a call to process remaining elements in the queue was added

After #9764, it seems to work, but synchronization is still fuzzy. For example, if the hasher calls `ReconnectNodeRemover.getRecordsToDelete()`, and the remover is closed in a parallel threads, the queue thread is stopped, but there are calls to `workQueue.pause()` and `workQueue.resume()` in `getRecordsToDelete()`. What do these calls do if the queue thread is stopped? Probably, no-ops + error log messages, probably not.

Is a separate queue thread needed in node remover at all? Can node requests be handled right on the learner thread? This ticket is to find answers to questions like this.

**Comment by artemananiev** [comment]
`ReconnectNodeRemover` has been re-worked completely as a part of #11231, it now doesn't use any queues or threads. This issue is no longer relevant.
